draft-kinnear-webtransport-http2.md

```diff
@@ -226,6 +226,10 @@ be sent on a stream in the "idle", "open", or "half-closed (remote)" state, see
 Like HEADERS, the CONTINUATION frame (type=0x9) is used to continue a sequence
 of header block fragments, if the headers do not fit into one WTHEADERS frame.
 
+Unlike HEADERS, the Header Block Fragment field MAY be empty.  There
+are no predefined semantics of the fields in the decoded block.  The
+application uses it to convey arbitrary structured metadata.
```

Comment:
If we're going to not define additional semantics here, should we indicate that there needs to be a replacement field defined by an extension for it to be considered valid? Or put in place some other extension mechanism? 

It seems like we jump into a whole additional layer of complexity by leaving it as an exercise for the implementor, is there a way to arrange this so it's very clear about what happens if you choose to leave it empty? 

(Having an empty WTHEADERS frame is totally reasonable and what we've actually implemented so far, so I'm not bothered by that, just want to be clear about what would happen if someone actually wanted to use "structured" metadata.)

Reply:
Of course if two implementations (client/server) want to interoperate, they need to agree on the semantics of the fields.  But they also need to agree on the content of the streams and/or datagrams, so I'm not sure it's totally different.  

Does the CONNECT handshake otherwise allow endpoints to negotiate what the semantics are for the WT session?

Maybe I can make it explicit in this PR:

```
There are no predefined semantics of the fields in the decoded block.  Applications can use it to convey arbitrary structured metadata.  Applications utilizing the header block SHOULD negotiate the semantics of the fields in the same manner they negotiate the content of streams or datagrams.
```